Modify Csharp Templates

DarrenSunandar opened the pull request:
# Description

When generating the C# wrapper for SplashKit, it would be nice for the translator to also generate C#'s XML based documentation comments from the existing HeaderDoc comments. I've firstly work on another pull request and from the review I put the pull request to the wrong repo, here's the link to the closed pull request: https://github.com/splashkit/splashkit-translator/pull/58


## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I had to modify the csharp templates using the ERB templating system. I modify the code so it correctly inputs the XML comments into splashkit.cs.

## Testing Checklist

- [ ] Tested with sktest
- [ ] Tested with skunit_tests

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from ... on the Pull Request

DarrenSunandar committed:
Modify Csharp Templates

Liquidscroll reviewed (commented):
Okay this is looking good! Here's a snipped of SplashKit.cs:

![image](https://github.com/user-attachments/assets/dd99af30-1c6b-4d14-b84b-b4244cf00460)

For now, we'll need to clarify whether we should be providing comments for functions like `FetchOrCreate` and after that I think it's ready to be approved.

Great work!

Liquidscroll commented on res/translators/csharp/SplashKit/classes/resource_classes.cs.erb:
```diff
@@ -3,15 +3,23 @@
 {
   private <%= class_name %>(IntPtr ptr) : base(ptr, true) {}
 
+  /// <summary>
+  /// Fetches or creates an instance of <%= class_name %> from the given pointer.
+  /// </summary>
+  /// <param name="ptr">The pointer to the SplashKit resource.</param>
+  /// <returns>An instance of <%= class_name %>.</returns>
```
Okay, so we should remove this as this is an internal function and our users can never call this, and so can never access this information.

Liquidscroll commented on res/translators/csharp/SplashKit/classes/resource_classes.cs.erb:
```diff
@@ -40,39 +52,54 @@ public static class <%= class_name %>
       fn = the_class[:destructor]
       method_data = get_method_data(fn)
 %>
+    /// <summary>
+    /// Frees the resources held by this instance of <%= class_name %>.
+    /// </summary>
```
Another internal function; this should be removed.

Liquidscroll reviewed (commented):
I've commented the pieces of code that should be removed. Once these are removed, this is ready for approval :)

DarrenSunandar committed:
Remove Internal Function

Liquidscroll reviewed (approved):
Looks good to me!

Great work with this! As far as I can tell the comments are correct and in the correct place, and no private or internal functions have comments.

hugglesfox reviewed (approved):
I don't know ruby so can't comment too much on the code however it creates the expected output so looks good to me!

WhyPenguins committed:
small output formatting fixes
(also, perhaps there's a better way to enforce the newlines after a
<%= %> other than just adding a blank line, but my knowledge of Ruby
is limited so hopefully this is alright for now)

WhyPenguins committed:
fix parameter case and remove `self` parameter from method comments

WhyPenguins committed:
remove remaining trailing '.'s
